src/coreclr/src/vm/gcenv.ee.cpp

```diff
@@ -22,6 +29,43 @@ void GCToEEInterface::SuspendEE(SUSPEND_REASON reason)
     _ASSERTE(reason == SUSPEND_FOR_GC || reason == SUSPEND_FOR_GC_PREP);
 
     g_pDebugInterface->SuspendForGarbageCollectionStarted();
+    
+    if (GetIntConfigValue("GCGenAnalysisGen", nullptr, &gcGenAnalysisGen))
+    {
+        if (GetIntConfigValue("GCGenAnalysisBytes", nullptr, &gcGenAnalysisBytes))
+        {
+            gcGenAnalysis = true;
+        }
+    }
+    if (gcGenAnalysis && event_pipe_state == 0)
+    {
+        event_pipe_state = 1;
+        LPCWSTR outputPath = nullptr;
+        outputPath = W("trace.nettrace");        
```

Comment:
Should this rather use regular event stream instead of ad-hoc file? Do we have ad-hoc file like this anywhere else?

Reply:
> Should this rather use regular event stream instead of ad-hoc file? 

The regular event pipe worked by having a client connecting to it, that gives us a socket where the events can write to. In this case, we do not have a client, therefore there is no socket, the only alternative is a file.

> Do we have ad-hoc file like this anywhere else?

We have a precedent of writing to a file here.

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/vm/eventpipe.cpp#L134-L147